### ISSUE TITLE
Cannot set timeouts over 15 seconds

### ISSUE THREAD

**Issue by Stono** [istio/istio#6230]
Hi, 
I have the following `VirtualService`:

```
  hosts:
  - app
  http:
  - route:
    - destination:
        host: app
      weight: 100
    timeout: 600.000s
```

But I'm getting:

```
< Content-Type: text/plain
< Content-Length: 24
< Connection: keep-alive
< x-envoy-upstream-service-time: 15038
< Strict-Transport-Security: max-age=15768000
* HTTP error before end of send, stop sending
<
* Closing connection 0
upstream request timeout%
```

When posting a file:

```
curl -v -F file=@moo http://app.vds-dvla-importer.svc.cluster.local/api/import
```

The timeout is definitely working, as if I lower it to say `timeout: 10.000s`, it errors after 10.

**Comment by ymesika** [comment]
Isn't it resolved by #6042 ?

**Comment by Stono** [comment]
Demonstrably not!

**Comment by Stono** [comment]
Closing this as @ymesika has rightly pointed out this is a known issue for which there is a bug fix in 0.8.1+

As a temporary workaround for anyone else finding themselves in this situation, adding the following:

```
spec:
  http:
  - appendHeaders:
      x-envoy-upstream-rq-timeout-ms: "0"
```

To your `VirtualService` acts as a work around

**Comment by msogrin** [comment]
@Stono Adding this resulted in a warning in ingressgateway, due to empty `weighted_clusters` block.
```
[2018-07-04 11:03:21.746][19][warning][upstream] external/envoy/source/common/config/grpc_mux_impl.cc:218] gRPC config for type.googleapis.com/envoy.api.v2.Listener update rejected: Error adding/updating listener 0.0.0.0_80: Proto constraint validation failed (HttpConnectionManagerValidationError.RouteConfig: ["embedded message failed validation"] | caused by RouteConfigurationValidationError.VirtualHosts[i]: ["embedded message failed validation"] | caused by VirtualHostValidationError.Routes[i]: ["embedded message failed validation"] | caused by RouteValidationError.Route: ["embedded message failed validation"] | caused by RouteActionValidationError.WeightedClusters: ["embedded message failed validation"] | caused by WeightedClusterValidationError.Clusters: ["value must contain at least " '\x01' " item(s)"]): stat_prefix: "http"
...
      route {
        weighted_clusters {
        }
        request_headers_to_add {
          header {
            key: "x-envoy-upstream-rq-timeout-ms"
            value: "0"
          }
        }
        use_websocket {
        }
      }
```
